tests/Unit/Domain/DomainCreators/Test_Rectangle.cpp

```diff
@@ -0,0 +1,129 @@
+  const Orientation<2> aligned_orientation{};
+
+  const DomainCreators::Rectangle rectangle{lower_bound, upper_bound,
+                                            std::array<bool, 2>{{false, false}},
+                                            num_elements, grid_points};
+  test_rectangle_construction(
+      rectangle, lower_bound, upper_bound, grid_points, num_elements,
+      std::vector<std::unordered_map<Direction<2>, BlockNeighbor<2>>>{{}},
+      std::vector<std::unordered_set<Direction<2>>>{
+          {{Direction<2>::lower_xi()},
+           {Direction<2>::upper_xi()},
+           {Direction<2>::lower_eta()},
+           {Direction<2>::upper_eta()}}});
+
+  const DomainCreators::Rectangle periodic_x_rectangle{
+      lower_bound, upper_bound, std::array<bool, 2>{{true, false}},
+      num_elements, grid_points};
+  test_rectangle_construction(
+      periodic_x_rectangle, lower_bound, upper_bound, grid_points, num_elements,
+      std::vector<std::unordered_map<Direction<2>, BlockNeighbor<2>>>{
+          {{Direction<2>::lower_xi(), {0, aligned_orientation}},
+           {Direction<2>::upper_xi(), {0, aligned_orientation}}}},
+      std::vector<std::unordered_set<Direction<2>>>{
+          {{Direction<2>::lower_eta()}, {Direction<2>::upper_eta()}}});
+
+  const DomainCreators::Rectangle periodic_y_rectangle{
+      lower_bound, upper_bound, std::array<bool, 2>{{false, true}},
+      num_elements, grid_points};
+  test_rectangle_construction(
+      periodic_y_rectangle, lower_bound, upper_bound, grid_points, num_elements,
+      std::vector<std::unordered_map<Direction<2>, BlockNeighbor<2>>>{
+          {{Direction<2>::lower_eta(), {0, aligned_orientation}},
+           {Direction<2>::upper_eta(), {0, aligned_orientation}}}},
+      std::vector<std::unordered_set<Direction<2>>>{
+          {{Direction<2>::lower_xi()}, {Direction<2>::upper_xi()}}});
+
+  const DomainCreators::Rectangle periodic_xy_rectangle{
+      lower_bound, upper_bound, std::array<bool, 2>{{true, true}}, num_elements,
+      grid_points};
+  test_rectangle_construction(
+      periodic_xy_rectangle, lower_bound, upper_bound, grid_points,
+      num_elements,
+      std::vector<std::unordered_map<Direction<2>, BlockNeighbor<2>>>{
+          {{Direction<2>::lower_xi(), {0, aligned_orientation}},
+           {Direction<2>::upper_xi(), {0, aligned_orientation}},
+           {Direction<2>::lower_eta(), {0, aligned_orientation}},
+           {Direction<2>::upper_eta(), {0, aligned_orientation}}}},
+      std::vector<std::unordered_set<Direction<2>>>{{}});
+}
+
+SPECTRE_TEST_CASE("Unit.Domain.DomainCreators.Rectangle.Factory",
+                  "[Domain][Unit]") {
+  test_factory_creation<DomainCreator<2, Frame::Inertial>>(
+      "  Rectangle:\n"
+      "    LowerBound: [0,0]\n"
+      "    UpperBound: [1,2]\n"
+      "    IsPeriodicIn: [True,False]\n"
+      "    InitialGridPoints: [3,4]\n"
+      "    InitialRefinement: [2,3]\n");
+}
+
+// [[OutputRegex, index = 1]]
```

Comment:
Check more of the ASSERT message, ie the condition that failed. I believe that is printed.

Reply:
I changed my mind on this: these functions are going away anyway, so all is good.